src/components/Dialogs/Dialog/Dialog.tsx

```diff
@@ -18,42 +21,66 @@ export interface DialogProps extends ModalProps {
     /** Dialog title */
     title?: string;
     /** Properties of the accept button */
-    acceptButton: ButtonProps;
+    acceptButton?: ButtonProps;
     /** Properties of the cancel button */
     cancelButton?: ButtonProps;
+    /** the context of the buttons in the footer */
+    variant?: Omit<ButtonContext, 'secondary'>;
+    /** closes the dialog */
+    onClose: () => void;
 }
 
 const { block, elem } = bem('Dialog', styles);
 
 export const Dialog: React.FC<DialogProps> = (props) => {
-    const { isOpen, children, title, acceptButton, cancelButton, ...rest } = props;
+    const {
+        isOpen,
+        children,
+        title,
+        acceptButton,
+        cancelButton,
+        variant = 'primary',
+        onClose,
+        ...rest
+    } = props;
 
+    const { ref, height } = useResizeObserver<HTMLDivElement>();
+    const showButtons = !!acceptButton;
     const isConfirm = !!cancelButton;
 
     return (
-        <Modal {...rest} {...block(props)} isOpen={isOpen}>
-            <div {...elem('content', props)} role="alert">
+        <Modal {...rest} {...block('dialog')} isOpen={isOpen}>
+            <div {...elem('header')} role="alert">
                 {!!title && (
                     <Heading level="h2" {...elem('title', props)}>
                         {title}
                     </Heading>
                 )}
+                <MdClose onClick={onClose} data-testid="close-icon" />
```

Comment:
I thought this will only be displayed when there is a header section, which will only be there if there is a title... Or do I see the design wrongly?

Reply:
I kept it because one of the cases is to use the dialog without any buttons which means that user won't have any close/cancel button on the dialog.

Reply:
Looking at figma, every variant has either Title + Close or Buttons. The whole header should be conditionally rendered based on the title being defined or not.

Reply:
This icon should be inside of a button otherwise the click area is very small and there is no indication that it is a button visually or with screen readers.
Also if it is a button you don't need to use data-testid to find it in the dom.